src/pages/saga.js

```diff
@@ -0,0 +1,35 @@
+//outsource dependencies
+import { toastr } from 'react-redux-toastr';
+import { call, put, takeEvery } from 'redux-saga/effects';
+//local dependencies
+import { TYPE } from './reducer';
+import { history } from '../history';
+import { ROUTES } from '../constants/route';
+import { localStorageMethods } from '../utils';
+import { instancePrivate } from '../services/api';
+
+export default function * initialize () {
+  yield takeEvery(TYPE.INITIALIZE, initializeSaga);
+}
+
+function * initializeSaga () {
+  const token = yield call(localStorageMethods.getToken);
+  try {
+    if (token?.data?.accessToken) {
+      yield put({ type: TYPE.META, payload: { token: token.data.accessToken } });
+      yield call(localStorageMethods.setToken, token);
+      instancePrivate.defaults.headers.common.Authorization = `Bearer ${token.data.accessToken}`;
+      yield call(history.push, ROUTES.USER);
+    } else {
+      yield call(history.push, ROUTES.SIGN_IN);
+    }
+  } catch (error) {
+    yield call(toastr.error, error.message);
+    yield put({ type: TYPE.META, payload: { errorMessage: error.message } });
+    yield call(history.push, ROUTES.SIGN_IN);
```

Comment:
```suggestion
    yield call(history.push, ROUTES.SIGN_IN);
```
Do it in axios interceptor if you wasn't able to refresh token